apps/api/pages/api/event-types/_post.ts

```diff
@@ -276,6 +281,11 @@ async function postHandler(req: NextApiRequest) {
 
   await checkPermissions(req);
 
+  if (isAdmin && parsedBody.parentId) {
+    await checkParentEventOwnership(parsedBody.parentId, userId);
+    await checkUserMembership(parsedBody.parentId, parsedBody.userId);
```

Comment:
We check:
1. If `parentId` provided is of the eventType that is a team event type, and the user making the request is either `Admin` or `Owner` of that team.
2. If `userId` provided is part of the team that the parentId belongs to.

Reply:
userId could be undefined. Causing a compile error.